Update for Olympia

DzhideX opened the pull request:
Changes:
- Set `@joystream/types` package version to `^0.18.1`
- Update `councilData` function to work with the new types (used [this code](https://github.com/Joystream/pioneer/blob/olympia/packages%2Fui%2Fsrc%2Fcouncil%2Fhooks%2FuseElectionStage.ts) as inspiration)
- Removed memo from `transfers.ts` and from the `Exchange` type in `db.ts`

Lezek123 reviewed (changes requested):
I haven't noticed this before, because the old Pioneer was using `balances.transfer` method for transfers, but both https://polkadot.js.org and CLI use `balances.transferKeepAlive` and those extrinsics are not currently recognized by the transactions processor. This needs to be fixed.

Another issue I noticed while testsing is that automatic burns seem to have stopped working:
```
2022-03-10 10:45:06        RPC-CORE: submitAndWatchExtrinsic(extrinsic: Extrinsic): ExtrinsicStatus:: 1010: Invalid Transaction: Inability to pay some fees , e.g. account balance too low
2022-03-10 10:45:06             DRR: 1010: Invalid Transaction: Inability to pay some fees , e.g. account balance too low
```
This is probably because in Olympia the _existential deposit_ of **10 JOY** was introduced, which prevents making a transaction with a tip that's > `current_balance - 10 JOY`

DzhideX commented:
I've added `transferKeepAlive` to the methods check so that now it is being recognized correctly. 

As for the `autoburn` issue, I fixed it in such a way where it always tries to keep a balance of `10 JOY`. This means, either the first burn would be for `10 JOY` less or how I actually intended it to be is that the one setting up the server needs to send `10 JOY` to it and then it should theoretically work without any problems. I don't know if you can think of a better solution for that?

DzhideX commented on src/joyApi.ts:
```diff
@@ -364,7 +364,19 @@ export class JoyApi {
+    // and we therefore don't want to try and burn the whole balance but rather try to keep the
+    // account balance at the EXISTENTIAL_DEPOSIT amount.
+
+    const EXISTENTIAL_DEPOSIT = 10;
```
Indeed, I wanted to ask if it was possible to get that data from the api directly but forgot. Thank you!

Lezek123 reviewed (approved):
LGTM

bwhm reviewed (approved):
:+1:
